text/0000-fabric-private-chaincode-1.0.md

```diff
@@ -0,0 +1,720 @@
+If the validation succeeds, the enclave credentials are stored on the ledger and thus available to channel members. 
+Note that the recommended FPC Enclave Registry endorsement policy is meant to protect the integrity of these checks and the credentials stored on the ledger.
+<!--  -->
+The enclave registration is illustrated in Step 10 - 12. in the figure above.
+This completes the deployment of a FPC Chaincode.
+
+Note that the deployment described here focuses on the first realization of FPC where a single enclave per FPC Chaincode is deployed. For future releases we intend to enhance FPC with multi-enclave support as also described in the roadmap section.
+In a multi-enclave deployment, the key material to encrypt/decrypt the chaincode state and the chaincode arguments must be shared among the FPC Chaincode enclaves.
+We already propose a key generation and distribution protocol as part of our [FPC specification](../images/fpc/full-detail/fpc-key-dist.png).
+
+## FPC Transaction Flow
+
+Now we describe the FPC transaction flow comprising client invocation, chaincode execution, and enclave endorsement validation as illustrated in the figure below. 
+
+![Transaction](../images/fpc/high-level/Slide4.png)
+
+### Client Invocation
+
+The client application performs an FPC Chaincode invocation through the FPC Client SDK.
+If not cached locally, the SDK will query first the Enclave Registry chaincode (at multiple peers) to retrieve the FPC chaincode's public encryption key.
+The public encryption key is used to encrypt the contents of the transaction proposal (i.e., the invocation arguments) before it is then sent as a query to the FPC Chaincode on the peer hosting the Enclave.
+In addition to the invocation arguments, the encrypted message contains a symmetric key generated by the SDK that is used later for the encryption of the transaction response.
+Moreover, with the help of the enclave registry, the SDK discovers the endpoints of the peers that host the FPC Chaincode.
+All these steps are fully transparent to the applications layer.
+
+The query is forwarded via the peer and some glue code to the FPC Shim inside the Enclave.
+Inside the Enclave, the FPC Shim decrypts the arguments of the proposal and saves the client's response encryption key.
+After that, it invokes the FPC Chaincode with the plaintext arguments.
+In the figure above, the invocation is illustrated with Step 1 - 5.
+
+### Chaincode Execution
+
+The FPC Chaincode processes the invocation according to the implemented chaincode logic.
+While executing, the chaincode can access the World State through `getState` and `putState` operations provided by the FPC Shim. From the chaincode perspective, the arguments of the former, and the output the latter, are plaintext data.
+
+The FPC Shim fetches the state data from the peer and loads it into the enclave, and similarly it stores data by forwarding it to the peer.
```

Comment:
Not clear what "stores data by forwarding it to the peer" means. Typically no data is stored during chaincode execution.

Reply:
I agree, this sentence is misleading. I've updated the description.